### PR TITLE
Upgrade tools and cert-manager images

### PR DESCRIPTION
**Pull request by inteon**
- bumped the versions in `modules/cert-manager/00_mod.mk` and `modules/tools/00_mod.mk`
- ran `make learn-tools-shas` and `make learn-image-shas` to learn the new shas.

### CODE REVIEW

File: scripts/learn_kind_images.helper.mk [hunk 1]
```diff
@@ -33,6 +33,9 @@ bin_dir := _bin
 $(bin_dir) $(bin_dir)/scratch:
 	mkdir -p $@
 
+HOST_OS := unused
+HOST_ARCH := unused
```
Comment (inteon) [inline scripts/learn_kind_images.helper.mk:37]: this fixes a warning message that the `make upgrade-kind-images` was outputting.

File: scripts/learn_kind_images.sh [hunk 2]
```diff
@@ -98,6 +98,7 @@ echo "" >> "${kind_versionfile}.tmp"
 
 release_json=$(curl -fsSL "https://api.github.com/repos/kubernetes-sigs/kind/releases/tags/${kind_version}"| jq '
   [ .body  | capture("- v?1\\.(?<minor>[0-9]+)(.(?<patch>[0-9]+))?: `kindest/node:v(?<version>[^@]+)@sha256:(?<sha256>[^`]+)`\r"; "g") ]
+  | group_by(.minor) | map(max_by(.patch))
```
Comment (inteon) [inline scripts/learn_kind_images.sh:101]: kind lists multiple patch versions for some of the minor releases.
We just select the latest (max) patch version for each.

**Comment by erikgb** [comment]
Nice, I was just looking into upgrading controller-gen when I saw this PR.

/lgtm

**Pull request merged**